app/src/main/java/com/celzero/bravedns/ui/NotificationHandlerDialog.kt

```diff
@@ -0,0 +1,193 @@
+        intent.flags = Intent.FLAG_ACTIVITY_SINGLE_TOP
+        intent.putExtra(Constants.VIEW_PAGER_SCREEN_TO_LOAD, FirewallActivity.FirewallTabs.ALL_APPS.screen)
+        startActivity(intent)
+        finish()
+    }
+
+    private fun handleAccessibilitySettings() {
+        val builder = AlertDialog.Builder(this)
+        builder.setTitle(R.string.alert_permission_accessibility_regrant)
+        builder.setMessage(R.string.alert_firewall_accessibility_regrant_explanation)
+        builder.setPositiveButton(
+            getString(R.string.univ_accessibility_crash_dialog_positive)) { _, _ ->
+            openRethinkAppInfo(this)
+        }
+        builder.setNegativeButton(
+            getString(R.string.univ_accessibility_crash_dialog_negative)) { _, _ ->
+            finish()
+        }
+        builder.setCancelable(false)
+
+        builder.create().show()
+    }
+
+    private fun openRethinkAppInfo(context: Context) {
+        val intent = Intent(Settings.ACTION_APPLICATION_DETAILS_SETTINGS)
+        val packageName = context.packageName
+        intent.data = Uri.parse("package:$packageName")
+        ContextCompat.startActivity(context, intent, null)
+    }
+
+    private fun showAppPauseDialog(trampolineType: TrampolineType) {
+        val builder = AlertDialog.Builder(this)
+
+        builder.setTitle(R.string.notif_dialog_pause_dialog_title)
+        builder.setMessage(R.string.notif_dialog_pause_dialog_message)
+
+        builder.setCancelable(false)
+        builder.setPositiveButton(R.string.notif_dialog_pause_dialog_positive) { _, _ ->
+            VpnController.getBraveVpnService()?.resumeApp()
+            when (trampolineType) {
+                TrampolineType.ACCESSIBILITY_SERVICE_FAILURE_DIALOG -> {
+                    handleAccessibilitySettings()
+                }
+                TrampolineType.NEW_APP_INSTAL_DIALOG -> {
+                    launchFirewallActivity()
+                }
+                else -> {
+                    // no-op
+                }
+            }
+        }
+
+        builder.setNegativeButton(R.string.notif_dialog_pause_dialog_negative) { _, _ ->
+            openAppPausedActivity()
+        }
+
+        builder.create().show()
+    }
+
+    private fun openAppPausedActivity() {
+        val intent = Intent()
+        intent.setClass(this, PauseActivity::class.java)
+        startActivity(intent)
+        finish()
+    }
+
+    private fun isAccessibilityIntent(intent: Intent): Boolean {
+        // Created as part of accessibility failure notification.
+        val extras = intent.extras
+        if (extras != null) {
+            val value = extras.getString(Constants.NOTIF_INTENT_EXTRA_ACCESSIBILITY_NAME)
+            if (!value.isNullOrEmpty() && value == Constants.NOTIF_INTENT_EXTRA_ACCESSIBILITY_VALUE) {
+                return true
+            }
+        }
+        return false
+    }
+
+    private fun isNewAppInstalledIntent(intent: Intent): Boolean {
```

Comment:
this code is repeated elsewhere too. move it to util?

Reply:
Removed the code from `FirewallActivity`